File: src/main/java/com/freya02/bot/versioning/maven/MavenBranchProjectDependencyVersionChecker.java
package com.freya02.bot.versioning.maven;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.nio.file.Path;

public class MavenBranchProjectDependencyVersionChecker extends MavenProjectDependencyVersionChecker {
	private final String targetBranch;

	public MavenBranchProjectDependencyVersionChecker(Path lastSavedPath, String ownerName, String artifactId, String targetArtifactId, String targetBranch) throws IOException {
		super(lastSavedPath, ownerName, artifactId, targetArtifactId);

		this.targetBranch = targetBranch;
	}

	@Override
	@NotNull
	protected String getTargetBranchName() {
		return targetBranch;
	}
}

File: src/main/kotlin/com/freya02/bot/versioning/maven/MavenBranchProjectDependencyVersionChecker.kt
package com.freya02.bot.versioning.maven

import com.freya02.bot.versioning.LibraryType
import java.nio.file.Path

class MavenBranchProjectDependencyVersionChecker(
    lastSavedPath: Path,
    githubOwnerName: String,
    githubRepoName: String,
    targetArtifactId: String,
    override val targetBranchName: String
) : MavenProjectDependencyVersionChecker(lastSavedPath, githubOwnerName, githubRepoName, targetArtifactId) {
    constructor(
        lastSavedPath: Path,
        project: LibraryType,
        targetArtifactId: String,
        targetBranchName: String
    ) : this(lastSavedPath, project.githubOwnerName, project.githubRepoName, targetArtifactId, targetBranchName)
}